src/Memory/InputQueue.ts

```diff
@@ -6,117 +6,149 @@ import * as BB from 'botbuilder'
 import { CLDebug, DebugType } from '../CLDebug'
 import InProcessMessageState from './InProcessMessageState'
 
-const MESSAGE_TIMEOUT = 10000
+// Two minutes after which we assume something went wrong in processing message
+const MESSAGE_TIMEOUT = 120000
 
 export interface QueuedInput {
+    conversationId: string
```

Comment:
>conversationId [](start = 4, length = 14)

was this source of the bug?

Reply:
No.  There used to be a single mutex/semphore and a single queue.  When memory was refactored, it was changed to a mutex per conversationId (which was the right thing to do), but the queue was still shared, meaning that when the mutex was freed it would grab the next conversation which could be for a different conversation not yet ready for another input